### DISCUSSION TITLE
Game logic

### DISCUSSION THREAD

**Discussion by fasteddys** [axuno/Volleyball-League#150] [General]
Dear Axuno, this is really cool, I am trying to use this in Cricket and lacrosse. 
Can you please share which classes have the game logic and team setup/matches?
There are two test projects/sql and I was trying to see why my build is failing in .NET 8, should I focus on one specifically?
Audits, we have to be careful on who is logging in actions, there are some free Open audit C# libraries can we use them.
Thanks
Please advise

**Comment by axunonb** [comment]
The project was created with only one kind of sport in mind, which is volleyball. That's why the the game logic is included in many classes. Team setup is not specific to volleyball, however, same as many other features like accounts, identity, backgroud task etc. In #139 a similar question was asked, also related to the build failing. Could you elaborate what exactly goes wrong?

**Comment by ridingmower** [comment]
Hi, I noticed you tagged this, just wanted to chime in, the error is mainly around the startup WebApplicationBuilder.WebHost, I managed to resolve it by upgrading to .NET 9 which also runs to 2026, not sure what the diff was with .NET 8 LTS

I like how you organized your code, its got focus on sports and solve one set of problems well, I am trying to repurpose to another game.

BTW I had a chance to see your calendar site and was curious why was that site different from this site, or is it the same. I thought events would go hand in hand.

**Comment by axunonb** [comment]
> TournamentCalendar

TournamentCalendar allows for sport clubs to advertise their volleyball tournaments, which usually are onetime events taking place on weekends. [This is the live website](https://volleyball-turnier.de/calendar) for it.

You are right, sport events are closely related to a league, if we think of each match in a league was such an event.
What we have on our wish list is managing players who would take part in a league match or in other tournaments, but also in training sessions.

**Comment by fasteddys** [comment]
If you have a wish list or a roadmap maybe I can help on my end with whatever I can do

**Comment by fasteddys** [comment]
Thanks man, the new build works much better, BTW .NET 8 was also a smooth upgrade and was easily able to do the nuget updates for the solution, you should check it out.

C

- [x] an you share a picture with the game/logic classes, so I can update as needed. 
- [x] suggestion, combine tournament scheduler with tournament manager?
- [x] I see a team manager, but a captain is missing are these the same thing? can we change the team size and ladder
- [x] The sql connection is described in app.config and tenant config, does having them in two places collide/override
- [x] Also I noticed you spun your own time span, check this out https://www.codeproject.com/Articles/168662/Time-Period-Library-for-NET lot easier to use , latest on [onGithub ](https://rb.gy/krvf60)
- [ ] ![image](https://github.com/axuno/Volleyball-League/assets/96149622/befce8e7-c5f9-42c8-a575-7ebd0f5bfb5e)
______

- [x] I see custom attributes, which ones, and when should we use this resource constraint , rather since some some attributes check if user is logged in, what does this check?

```
[Route(TenantRouteConstraint.Template + "/team-application")]
[ControllerAttributes.TeamApplicationAllowed] // if not allowed, all controller actions are redirected to action "list"
[Authorize]
public class TeamApplication : AbstractController
```

**Comment by axunonb** [comment]
Hi
1. Re Picture with the game/logic classes: I understand this could be helpful, but time consuming. Most of the logic is implemented in the TournamentManager project. The League project is kind of the web frontend.
2. Re combine tournament manager with scheduler. Actually, the scheduler is included in namespace `TournamentManager.Plan` of the TournamentManager project.
3. Re  team manager vs captain. At the very beginning I used the term captain. But the captain is not necessarily also a team manager in terms of the software. Let's say, it's just a question of the term to use.
4. Re App.config: You're referring to TournamentManager.DAL.DatabaseSpecific.App.config. Not sure, but I think this file is auto-generated, by LlblGenPro. The connection string in use is in `Credentials.*.json`
5. `TimePeriod` (more related to time period than time span!) [TimePeriodLibrary](https://github.com/Giannoudis/TimePeriodLibrary) is younger than League and seems to have more under hood, than ours. I'll have a closer look. Thanks for the hint.
6. The `ControllerAttributes.TeamApplicationAllowed` is used on the `TeamApplication` controller class. It just makes sure that no team applications are possible while there is no new Tournament defined for the next season, so that application would go to the running tournament. It applies for all users.

**Comment by fasteddys** [comment]
is it possible to skip LlblGenPro, or get some promo under this use for this project.

**Comment by axunonb** [comment]
> is it possible to skip LlblGenPro, 

Generally speaking, it is, but this means a lot of work.

> or get some promo under this use for this project.

We asked the vendor few years ago. At this time, they didn't support OSS projects. Maybe it makes sense reaching out to them again...

**Comment by fasteddys** [comment]
Can we replace LLbeanPro with https://marketplace.visualstudio.com/items?itemName=michaelsawczyn.EFDesigner2022 this is free and more robust

**Reply by axunonb** [reply]
Sorry, I oversaw your latest question.
 [LLbeanPro](https://marketplace.visualstudio.com/items?itemName=michaelsawczyn.EFDesigner2022) looks like a nice tool, although it does not support a database-first approach.
Still, to move away from LlblGenPro means a lot of work. Did you start working on that, or did your drop the idea using Volleyball-League as a base for your project?
